client.go

```diff
@@ -66,6 +70,23 @@ type Client struct {
 	Options []ClientOption
 }
 
+// ProcessUmask contains the process wide umask set at init time. see client_unix.go
+var ProcessUmask os.FileMode = 0022
```

Comment:
I think I'd prefer if this was unexported. If you want to set the umask for go-getter you should set it on the `Client` directly. Package-variable global configuration is somewhat idiomatic (used a lot in the stdlib) but just always felt like a security risk or race waiting to happen.

Reply:
Note also: if we unexport it today, then if there are use cases we require it exported in the future we can always export it. But if we don't need it today, we can never really unexport it w/o breaking API guarantees. So I think its best to unexport it.

Reply:
That makes sense to me.

Reply:
Followup here, reading the `syscall.Umask` has been eliminated, our default umask is actually just 0, since the backwards compatibility preserving mode is to run like `tar --preserve-permissions`. So this eliminates the windows compatibility problem and this variable completely.